DNN Platform/Library/Entities/Modules/Settings/SettingsRepository.cs

```diff
@@ -63,23 +63,16 @@ public void SaveSettings(ModuleInfo moduleContext, T settings)
 
                         if(string.IsNullOrEmpty(settingValueAsString))
                         {
-                            if (settingValue is DateTime)
-                            {
-                                settingValueAsString = ((DateTime)settingValue).ToString("u");
-                            }
-                            else
-                            {
-                                settingValueAsString = settingValue.ToString();
-                            }
+                            settingValueAsString = GetSettingValueAsString(settingValue);
```

Comment:
Right ... it's all about culture. So as soon as the thread culture changes you're in big trouble. This can happen for many reasons. 

This code is mine. The only robust method serializing dates is to use the universal format. I was disappointed that the original code just relied on the core to serialize. This leads to very awkwardly formatted dates which will fail in deserialization when the locale changes. That is not good behavior as the (de)serialization needs to be culture agnostic.